sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -984,7 +984,8 @@ case class Aggregate(
 
   // Whether this Aggregate operator is group only. For example: SELECT a, a FROM t GROUP BY a
   private[sql] def groupOnly: Boolean = {
-    aggregateExpressions.map {
+    // aggregateExpressions can be empty through Dateset.agg
+    aggregateExpressions.nonEmpty && aggregateExpressions.map {
```

Comment:
is it more explicit to check `groupingExpressions.nonEmpty`? Logically, an aggregate operator without output columns is still group only if it has group columns.

Reply:
make sense